### PR TITLE
Variable number of levels

### PR DESCRIPTION
**Pull request by mariana-s-fernandes**
Added support for variable number of levels. This option is only available for the library mode and not in the plugin (which continues to accept the fixed number of 3 levels).

### CODE REVIEW

**Comment by mariana-s-fernandes** [comment]
@abey79 Can you please test this with some examples of your own and see if everything works?

**Comment by jimmykl** [comment]
Hi @mariana-s-fernandes 

I would like to test this but have only used the plugin version before and when I try following the instructions for the library `python ./examples/skull.py` gives me `ModuleNotFoundError: No module named 'hatched'`

This is the process I used to install/setup:
```
git clone https://github.com/plottertools/hatched.git
cd hatched
git fetch origin pull/14/head:MASTER
python3 -m venv venv
source venv/bin/activate
pip install -r requirements.txt

**Comment by mariana-s-fernandes** [comment]
Hi @jimmykl , my changes should not have impacted that process. But this error has to do with the weird way that Python deals with paths. I don't remember the exact right way to deal with this (I think there is something that needs to be added to the PYTHONPATH env variable or something like that), but in the short term, copy the skull.py out of the examples folder and it should work.

File: hatched/hatched.py [hunk 1]
```diff
@@ -183,86 +184,93 @@ def _load_image(
...
+        all_lines.extend([ls for ls in mls[i].geoms])
+
+    return (MultiLineString(all_lines), *contours)
 
 
 def hatch(
     file_path: str,
     hatch_pitch: float = 5,
-    levels: Tuple[int, int, int] = (64, 128, 192),
+    levels: Union[int, Tuple[int]] = (64, 128, 192),
```
Comment (abey79) [inline hatched/hatched.py]: This should read `Tuple[int, ...]` to suggest an arbitrary number of int is acceptable.

File: hatched/hatched.py [hunk 2]
```diff
@@ -1,15 +1,16 @@
 import math
 import os
 import random
-from typing import Any, Iterable, Tuple
+from typing import Tuple, Iterable, Any, Union
 
 import cv2
 import matplotlib.collections
 import matplotlib.pyplot as plt
+import matplotlib.colors as mcolors
```
Comment (abey79) [inline hatched/hatched.py:8]: Please run `isort` to fix import order

**Comment by abey79** [comment]
Looks good to me, please check my two comment.

File: hatched/hatched.py (hunk 2 above)
Comment (mariana-s-fernandes) [reply hatched/hatched.py:8]: Thanks, fixed!

File: hatched/hatched.py (hunk 1 above)
Comment (mariana-s-fernandes) [reply hatched/hatched.py]: Fixed

**Comment by mariana-s-fernandes** [comment]
hi @abey79 , fixed the comments above

**Comment by abey79** [comment]
Looks good, thanks!

**Pull request merged**